packages/flutter_tools/test/general.shard/commands/build_aar_test.dart

```diff
@@ -107,6 +108,95 @@ void main() {
     });
   });
 
+  group('flag parsing', () {
+    Directory tempDir;
+    MockAndroidBuilder mockAndroidBuilder;
+
+    setUp(() {
+      mockAndroidBuilder = MockAndroidBuilder();
+      tempDir = globals.fs.systemTempDirectory.createTempSync('flutter_tools_build_aar_test.');
+    });
+
+    tearDown(() {
+      tryToDelete(tempDir);
+    });
+
+    testUsingContext('defaults', () async {
+      final String projectPath = await createProject(tempDir,
+        arguments: <String>['--no-pub']);
+      await runCommandIn(projectPath);
+
+      final Set<AndroidBuildInfo> androidBuildInfos = verify(mockAndroidBuilder.buildAar(
+        project: anyNamed('project'),
+        target: anyNamed('target'),
+        androidBuildInfo: captureAnyNamed('androidBuildInfo'),
+        outputDirectoryPath: anyNamed('outputDirectoryPath'),
+        buildNumber: '1.0',
+      )).captured[0] as Set<AndroidBuildInfo>;
+
+      expect(androidBuildInfos.length, 3);
+
+      final List<BuildMode> buildModes = <BuildMode>[];
+      for (final AndroidBuildInfo androidBuildInfo in androidBuildInfos) {
```

Comment:
Possibly overly pedantic, but It would be nice to separate the control flow from the expectations here. You could nest `contains` and `predicate`/`isA<>` matchers to accomplish this, though I don't know if its an improvement

Reply:
I tried it, I think it's harder to read...